provider/maas/environ_whitebox_test.go

```diff
@@ -217,6 +217,40 @@ func (suite *environSuite) TestStartInstanceStartsInstance(c *gc.C) {
 	c.Check(err, jc.Satisfies, errors.IsNotFound)
 }
 
+func (suite *environSuite) TestStartInstanceReturnsHostnameAsDisplayName(c *gc.C) {
```

Comment:
You don't need a whole new test here - just add an extra assert to TestStartInstanceStartsInstance or similar

Reply:
I originally thought that too. The reason that I've needed the extra test is that `TestStartInstanceStartsInstance` doesn't use a method that exposes `StartInstanceResult`. That's alluded to within the comments of the new test:

>     // Create controller. We're not doing testing StartInstance on node0 because
>     // env.ControllerInstances and many of juju/testing/instance helper
>     // functions don't provide us with access to the environs.StartInstanceResult

Reply:
It's easy to fix though. Instead of the existing test calling AssertStartInstance() helper, just break out the 3 lines inside that method and use in the test. Better to do that than add the extra LOC in a whole new test

Reply:
If you drill down a couple of API calls, the start instance result is available to use.
AssertStartInstance -> StartInstance -> StartInstanceWithConstraints -> StartInstanceWithParams

The latter returns a result struct. It's just a bit of tweaking to make it work.

Reply:
Sorted